src/ape_ethereum/ecosystem.py

```diff
@@ -88,7 +88,7 @@ class Config:
         smart_union = True
 
     @validator("gas_limit", pre=True)
-    def validate_gas_limit(cls, value: GasLimit) -> Union[Literal["auto", "max"], int]:
+    def validate_gas_limit(cls, value: GasLimit) -> GasLimit:
```

Comment:
Technically the input and output types are different since the output is only `Union[Literal["auto", "max"], int]` 

Does that need to be reflected here with perhaps two type definitions?

Reply:
I think it is ok to share the 1 type